lib/internal/test_runner/test.js

```diff
@@ -552,23 +552,20 @@ class Suite extends Test {
     try {
       const context = { signal: this.signal };
       this.buildSuite = this.runInAsyncScope(this.fn, context, [context]);
+      if (isPromise(this.buildSuite)) {
+        this.buildSuite = this.buildSuite.catch((err) => {
+          this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
+        });
+      }
```

Comment:
@aduh95 what do you think about this instead?:
```suggestion
      this.buildSuite = PromisePrototypeCatch(PromiseResolve(this.runInAsyncScope(this.fn, context, [context])), (err) => {
        this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
      });
```

Reply:
Yes that sounds good, it's also closer to how async/await is spec'd so even better.

`%Promise.prototype.catch%` looks up the `then` property of the promise, so it should instead be:

```suggestion
      this.buildSuite = PromisePrototypeThen(
        PromiseResolve(this.runInAsyncScope(this.fn, context, [context])),
         undefined,
         (err) => {
            this.fail(new ERR_TEST_FAILURE(err, kTestCodeFailure));
          }
        );
```

Reply:
that is what `PromisePrototypeCatch ` does:
https://github.com/nodejs/node/blob/3fb57843de58e1ee8bae72da74b6ad9e17de177b/lib/internal/per_context/primordials.js#L412-L413

Reply:
It won't for long, see https://github.com/nodejs/node/pull/43849.